### ISSUE TITLE
Introduce phase_lock_angle param to control msg header timestamp

### ISSUE THREAD

**Issue by mpitropov** [ros-drivers/velodyne#242]
**Is your feature request related to a problem? Please describe.**
Within the driver the packet message is published using the timestamp of the last packet. This means that the published pointcloud has the timestamp of the last packet. Also the transformed pountcloud is corrected to the timestamp of the last packet.
https://github.com/ros-drivers/velodyne/blob/7ecdf17173a609234df4e61cdd443748ccf7ab8f/velodyne_driver/src/driver/driver.cc#L243
This creates a problem with time sync where someone has a PPS signal triggering other sensors and also has a phase lock angle for their lidar. The lidar will have a delayed timestamp compared to other sensors.

**Describe the solution you'd like**
I think I'll have to create a param called phase_lock_angle which should be the same angle you have set in your lidar configuration. Adding to the current cut_angle loops seems to be the way to go although the double while loop is looking a bit large. When the packet angle passes the phase_lock_angle I will store the packet timestamp to be used for the published message header.

**Describe alternatives you've considered**
I have the cut angle set to behind the car and the phase lock set to in front of the car so in theory I would be able to just grab the timestamp of the center packet. Other people may have different setups though.

**Additional context**
![IMG_20190514_112242](https://user-images.githubusercontent.com/3324075/57710387-dd084c00-763a-11e9-9e15-eb9473c2282d.jpg)

**Comment by JWhitleyAStuff** [comment]
I agree with the general concept. Would you be willing/able to put in a Pull Request with this implemented?
